### PR TITLE
Add side effect for async migrations

### PR DESCRIPTION
**Pull request by fuziontech**
Needed to apply side effects during migrations that may be needed

## Changes

add side effect method

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: posthog/async_migrations/test/test_runner.py [hunk 1]
```diff
@@ -46,10 +48,14 @@ def test_run_migration_in_full(self):
         self.assertEqual(sm.posthog_min_version, "1.0.0")
         self.assertEqual(sm.posthog_max_version, "100000.0.0")
         self.assertEqual(sm.finished_at.day, datetime.today().day)
+        self.assertEqual(self.migration.sec.side_effect_count, 3)
+        self.assertEqual(self.migration.sec.side_effect_rollback_count, 0)
 
     @pytest.mark.ee
     def test_rollback_migration(self):
 
+        self.migration.sec.reset_count()
```
Comment (yakkomajuri) [inline posthog/async_migrations/test/test_runner.py:57]: could add this in the `setUp` function

File: posthog/async_migrations/utils.py [hunk 2]
```diff
@@ -12,12 +12,16 @@
...
+    """
     sql = op.rollback if rollback else op.sql
     if op.database == AnalyticsDBMS.CLICKHOUSE:
         execute_op_clickhouse(sql, query_id, op.timeout_seconds)
-        return
-
-    execute_op_postgres(sql, query_id)
+    else:
+        execute_op_postgres(sql, query_id)
+    return
```
Comment (yakkomajuri) [inline posthog/async_migrations/utils.py:24]:
```suggestion
```

**Review by yakkomajuri — APPROVED** [review]
lgtm - I'll get in the renaming PR and then we can get this in

File: posthog/async_migrations/runner.py [hunk 3]
```diff
@@ -111,6 +111,7 @@ def run_async_migration_next_op(migration_name: str, migration_instance: Optiona
 
     try:
         execute_op(op, current_query_id)
+        op.side_effect()
```
Comment (tiina303) [inline posthog/async_migrations/runner.py:114]: why after the op?

**Review by tiina303 — COMMENTED** [review]
Put up an alternative RFC https://github.com/PostHog/posthog/pull/7711 that I think would be simpler to use/understand (e.g. don't need to check the code to verify that side_effects were ran after the op vs before) and allow us more flexibility in the future

**Pull request merged**